src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs

```diff
@@ -80,8 +80,7 @@ public static void ReadAfterInitializationWithUriOnAsyncReaderTrows()
         {
             using (XmlReader reader = XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = true }))
```

Comment:
I'd suggest making this a Theory, where the async bool value is passed in as an argument to the test.  Ideally the behavior this is testing of Read would be the same regardless of the Async configuration.

Reply:
Does the same also apply to all the other tests in this file?

Reply:
I think it'd be a nice way to help verify the behavior is consistent.

Reply:
Actually, the behavior is not consistent: `Async = false` throws from `XmlReader.Create()`, not from `reader.Read()`, because of this code:

https://github.com/dotnet/corefx/blob/ac3be14326004969a24204226ffd4d6beafa2b59/src/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs#L608-L612

So, I used Theory on the two tests where it made sense and also created a new Fact for the `Async = false` version of this test.

Reply:
Ugh.  Ok, thanks.